tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Desugarer.scala

```diff
@@ -168,16 +168,16 @@ class Desugarer(gen: UniqueNameGenerator, tracker: TransformationTracker) extend
         case hd :: Nil =>
           val uniqueName = gen.newName()
           // LET tmp == [ fun EXCEPT ![i_n] = e ] IN
-          val funT = fun.typeTag.asTlaType1()
+          val funT = fun.typeTag.asTlaType1() // either FunT1, RecT1, TupT1, or SeqT1
           val operT = OperT1(Seq(), funT)
           val decl = tla
-            .declOp(uniqueName, tla.except(fun, hd, newValue) ? "f")
-            .typedOperDecl(Map("f" -> funT, "F" -> operT), "F")
+            .declOp(uniqueName, tla.except(fun, tla.tuple(hd) ? "t", newValue) ? "f")
+            .typedOperDecl(Map("f" -> funT, "t" -> TupT1(hd.typeTag.asTlaType1()), "F" -> operT), "F")
           Seq(decl)
```

Comment:
Looks like the code here would be shorter if you just just used the `typed` suffix, no?

Reply:
You are right. Fixed that.